Kernel/Arch/aarch64/RPi/Timer.cpp

```diff
@@ -186,13 +186,34 @@ static constinit Array const compatibles_array = {
 
 DEVICETREE_DRIVER(BCM2835TimerDriver, compatibles_array);
 
+// https://www.kernel.org/doc/Documentation/devicetree/bindings/timer/brcm,bcm2835-system-timer.txt
 ErrorOr<void> BCM2835TimerDriver::probe(DeviceTree::Device const& device, StringView) const
 {
+    if (DeviceTree::get().is_compatible_with("raspberrypi,4-model-b"sv))
+        return ENOTSUP; // HACK: The Pi 4 system timer doesn't appear to work on QEMU; only the generic ARM timer does.
```

Comment:
But both work on the actual hw?

Reply:
> But both work on the actual hw?
I can't remember if it tested it, but I would expect it to work as the broadcom doc mentions this timer as well.

The device is marked as `status = "disabled";` both for the pi 3 and 4. But we currently ignore that property completely. If we would respect that property we would also have to use the generic arm timer for the pi 3. But that timer is connected to another interrupt controller (which is actually the root interrupt controller), which we don't support.

> Can we check for qemu here as well?
ARM systems afaict don't have any hypervisor identification registers.